src/algorithm/MinimumBoundingCircle.cpp

```diff
@@ -299,23 +299,23 @@ MinimumBoundingCircle::pointWitMinAngleWithX(std::vector<Coordinate>& pts, Coord
 Coordinate
```

Comment:
Was the dereference even necessary in the first place, or should the function just return `const Coordinate*` ?

Reply:
You are right, in fact now that I'm looking at it there is a way less intrusive way to handle this which is to just initialize minAngPt to a pointer to the first element, instead of nullptr.

Reply:
You are absolutely right. All the other functions do the same, should I change the 3 of them?

Reply:
Now that I look into it, this would involve major changes in the API, changing `extremalPts` to an array of references / pointers and so on, so I'd rather do it in a different PR and leave this one with the copy for now.

Reply:
Sounds good. While you're in there, the other function arguments should probably be `const Coordinate &` instead of mutable...